website/blog/how-to-migrate-an-ios-app-to-bazel.md

```diff
@@ -0,0 +1,1470 @@
+
+You will need to set the `infoplists` and `entitlements` attributes to your `Info.plist` and `.entitlements` files,
+the `bundle_id` attribute to the value of the `PRODUCT_BUNDLE_IDENTIFIER` build setting,
+and the `minimum_os_version` attribute to the value of `IPHONEOS_DEPLOYMENT_TARGET` build setting.
+
+:::note
+
+**rules_apple** processes `Info.plist` and `.entitlements` files in a slightly different manner than Xcode.
+In Xcode,
+you have build settings such as
+`PRODUCT_BUNDLE_IDENTIFIER`,
+`CURRENT_PROJECT_VERSION`,
+and `PRODUCT_MODULE_NAME`.
+And when it processes an `Info.plist`,
+it substitutes references to those build settings with their resolved value.
+**rules_apple** doesn't use build settings,
+so it's not able to do the same substitutions.
+
+There are some attributes on bundling rules that **rules_apple** [will substitute][rules_apple-variable-subsitution] in place of a build setting,
+such as `bundle_id` for `PRODUCT_BUNDLE_IDENTIFIER`,
+but for the most part when using Bazel you'll need to remove these build setting references from `Info.plist` and `.entitlements` files.
+
+[rules_apple-variable-subsitution]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/common_info.md#variable-substitution
+
+:::
+
+Since the Mastodon iOS project doesn't use resource bundles,
+resources will be referenced directly with the `data` attribute of the static library target for the primary module.
+
+I adjusted the `Info.plist`s for the app extension targets and added `BUILD` files for them:
+
+```python
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_extension")
+load("@build_bazel_rules_apple//apple:resources.bzl", "apple_intent_library")
+load("@build_bazel_rules_swift//swift:swift.bzl", "swift_library")
+
+ios_extension(
+    name = "WidgetExtension",
+    bundle_id = "org.joinmastodon.app.WidgetExtension",
+    entitlements = "WidgetExtension.entitlements",
+    families = [
+        "ipad",
+        "iphone",
+    ],
+    infoplists = ["Info.plist"],
+    minimum_os_version = "16.0",
+    resources = glob(
+        [
+            "**/*.lproj/**",
+            "**/*.js",
+            "**/*.xcassets/**",
+        ],
+        exclude = [".*"],
+    ),
+    visibility = ["//visibility:public"],
+    deps = [":WidgetExtension.library"],
+)
+
+swift_library(
+    name = "WidgetExtension.library",
+    srcs = glob(["**/*.swift"]) + [":WidgetExtension.intent"],
+    module_name = "WidgetExtension",
+    deps = [
+        "@swiftpkg_lightchart//:LightChart",
+        "@swiftpkg_mastodonsdk//:MastodonSDKDynamic",
+    ],
+)
+
+apple_intent_library(
+    name = "WidgetExtension.intent",
+    src = "Base.lproj/WidgetExtension.intentdefinition",
+    language = "Swift",
+    visibility = ["//visibility:public"],
+)
+```
+
+:::note
+
+We are using [`glob`][glob] to collect our source files.
+This requires that none of the source files are unused.
```

Comment:
I don't get what this is referencing. Maybe it can be reworded to make more sense to me/others.